pkg/controllers/provisioning/launcher.go

```diff
@@ -111,6 +117,14 @@ func (l *Launcher) bind(ctx context.Context, node *v1.Node, pods []*v1.Pod) (err
 	return nil
 }
 
+func (l *Launcher) verifyResourceLimits(ctx context.Context, provisioner *v1alpha5.Provisioner) error {
+	latest := &v1alpha5.Provisioner{}
+	if err := l.KubeClient.Get(ctx, client.ObjectKeyFromObject(provisioner), latest); err != nil {
+		return err
```

Comment:
Are you familiar with the error wrapping/annotation pattern?

```suggestion
		return fmt.Errorf("getting current resource usage, %w", err)
```